### ISSUE TITLE
Derivatives - why global mesh?

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#596]
Should derivatives not default to there own mesh to take derivatives?

Currently:
https://github.com/boutproject/BOUT-dev/blob/master/src/sys/derivs.cxx#L87

I think something like:
```
const Field3D DDY(const Field3D &f, CELL_LOC outloc, DIFF_METHOD method) {
  return f.getMesh()->indexDDY(f,outloc, method) / f.getMesh()->coordinates()->dy;
}
```
and
```
Mesh * Field3D::getMesh(){
  if (fieldmesh){
    return fieldmesh;
  } else {
    return mesh;
  }
}
```
I hope this helps allowing several meshes 👍

**Comment by bendudson** [comment]
Yes, that would be much better. Hopefully we can find a way to eliminate the global mesh pointer

**Comment by dschwoerer** [comment]
closed by #598
